src/Compilers/VisualBasic/Portable/Symbols/ArrayTypeSymbol.vb

```diff
@@ -314,34 +314,36 @@ Namespace Microsoft.CodeAnalysis.VisualBasic.Symbols
         Friend MustOverride Overrides Function InternalSubstituteTypeParameters(substitution As TypeSubstitution) As TypeWithModifiers
 
         Public Overrides Function Equals(obj As Object) As Boolean
+            Return IsSameType(obj, TypeCompareKind.ConsiderEverything)
+        End Function
+
+        Friend Function IsSameType(obj As Object, compareKind As TypeCompareKind) As Boolean
+            Debug.Assert((compareKind And Not (TypeCompareKind.IgnoreCustomModifiersAndArraySizesAndLowerBounds Or TypeCompareKind.IgnoreTupleNames)) = 0)
```

Comment:
>TypeCompareKind.IgnoreCustomModifiersAndArraySizesAndLowerBounds Or TypeCompareKind.IgnoreTupleNames [](start = 47, length = 100)

Consider adding AllValidIgnoreOptionsForVB (or something similar) to TypeCompareKind enum.